tuf/data/roles.go

```diff
@@ -96,6 +97,16 @@ func IsBaseRole(role string) bool {
 	return false
 }
 
+// IsWildDelegation determines if a role represents a valid wildcard delegation
+// path, i.e. targets/*, targets/foo/*.
+// The wildcard may only appear as the final part of the delegation and must
+// be a whole segment, i.e. targets/foo* is not a valid wildcard delegation.
+func IsWildDelegation(role string) bool {
+	base := filepath.Dir(role)
```

Comment:
Do we also want to add a check that the last character of the role is `*`?  `path.Base(role)` will trim off any trailing slashes.